### DISCUSSION TITLE
Is it possible to run Webots without the GUI at all?

### DISCUSSION THREAD

**Discussion by chengliu-LR** [cyberbotics/webots#5007] [Q&A]
Hi all, currently I am trying to use Webots as my training environment for reinforcement learning agents on a `red hat linux` cluster (only terminal available). So I am wondering is it possible to run Webots without the GUI at all?

**Comment by omichel** [comment]
It is possible to run Webots in batch mode (e.g., with minimized windows) or on some off-screen virtual display. Since Webots relies on OpenGL for some device rendering (cameras, lidars, range-finders, etc.), it needs a connection with OpenGL, hence a display (which may be virtual). This means that your cluster machines should have a display installed (possibly virtual, like Xvfb). More information is available [here](https://stackoverflow.com/questions/58208419/does-webots-have-headless-mode). Another solution is to run Webots in a docker in headless mode, see documentation [here](https://www.cyberbotics.com/doc/guide/installation-procedure#run-webots-in-docker-in-headless-mode).
